cmd/guardian/main.go

```diff
@@ -135,6 +141,33 @@ func main() {
 
 func realMain(ctx context.Context) error {
 	setLogEnvVars()
+
```

Comment:
Should we move this into a helper function (like we did with `setLogEnvVars`) just to keep main clean?

Reply:
Probably, the main issue is handling the defer statements in the correct order, we would need to return a function for the caller to defer. I'll think about if metricswrap is really needed.

Reply:
Was able to factor some of the setup out.